18/URL scheme: draft (#159)
status-im/specs: docs/draft/17-url-scheme.md

---
permalink: /spec/17
parent: Draft specs
title: 17/URL SCHEME
---

# 17/Defined Url Scheme

> > Version: 0.1.0

Reviewer: If my previous suggestions are applied and since my understanding is that our clients are already adhering to this open PR to some extend:

```suggestion
> Version: 0.2.0
```

>
> Status: Draft
>
> Authors: Aleksandar Djenic <aleksandardjenic@status.im> Michal Iskierko <michal@status.im>
>

## Table of Contents

 1. [Url Scheme](#url-scheme)
 2. [Defined url](#defined-url)

## Url scheme

> - internal: "status-app://"
> - external: "https://status.app"

Reviewer: > Yes, both desktop and mobile should open and handle links starting with https:// , it's really important that they do this. – https://github.com/status-im/status-web/issues/327#issuecomment-1383866949

```suggestion
- universal: "https://status.app"
```

Reviewer: In the "Suggested change" above I proposed removal of `status-app://` scheme since I couldn't think of a scenario where it'd be necessary if support for universal links with `https://` was ensured.

However, [this](https://www.figma.com/file/Nt3KJ8AXm0VPpIRLxaHnw0/Sharing-for-Mobile?node-id=2370%3A299391&t=Pi7DGeQEZ1wjQHgQ-0) design, in case of

- user not having the app installed
- clicking install on the website
- going to the app store and installing it
- and then if ever going back to the website

requires

- "Join community in Status" -like buttons to trigger opening the now possibly installed app with provided context
  - using "possibly" here, because I'm not aware of reliable way to tell if the user actually installed the app
  - nor if "context" could be passed to the app if installed and opened from the app store
  - and "why" to keep the both schemes, because I don't think a reload of the website would trigger the action
    - I tried that in Safari on iOS which didn't work, but maybe some silent programatic action with `referrerpolicy` or else could achieve that
      - for repro, Long press [this](https://join.status.im/u/0x04ac78d5b61ea49eab8849826c628e01aed9d38ef8abe2fe53444e47bec31788049a5099ebf6dbbe5bc1c39fffee1e012faf201e3775296bec0aa38a5ddcdf1df7) link in Notes > Open in Safari (instead of Open in "Status") > Reload

@prichodko @jkbktl do you agree with keeping both schemes for the reasons stated above? Are you aware of a workaround which could be used if only `https://` was kept? Or should we create an issue to research that?

Reviewer: Custom scheme is still required, see https://github.com/status-im/status-desktop/issues/9549#issuecomment-1540801424.

## Defined url

| Name | Url | Description |
| ----- | ---- | ---- |

> | User profile | `/u/[compressed_user_key or ens_name]` | Display user profile popup for user with `compressed_user_key` or `ens_name` |

Reviewer: Specify public keys and signatures as URL fragments, placing them after `#`. In order to limit their exposure to hosting servers.

```suggestion
| User profile | `/u/<data>#<signature>` | Display user |
| | `/u#<compressed_user_public_key>` | |
| | `/u#<ens_name>` | |
```

> | Community |	`/c/[compressed_community_key]` | Open community with `compressed_community_key` |

Reviewer: ```suggestion
| Community |	`/c/<data>#<signature>` | Open community |
| | `/c#<compressed_community_public_key>` | |
```

> | Community channel | `/cc/[compressed_channel_key]`| Open community which has a channel with `compressed_channel_key` and makes that channel active |

Reviewer: ```suggestion
| Community channel | `/cc/<data>#<signature>`| Open community channel |
| | `/cc/<channel_uuid>#<compressed_community_public_key> | |
```

Reviewer: Await https://github.com/status-im/specs/pull/169#discussion_r1122122684

| Browse | `/b/[url]` |  Open `url` in the app's browser |

> | Post in channel | `/cc/[channel_key]/[message_key]` | Go to a message `message_key` in the channel `channel_key`

Reviewer: ```suggestion
| Post in channel | `/cc/<channel_uuid>/<message_id>#<compressed_community_public_key>` | Go to message in channel
```

Reviewer: Addressed at https://github.com/status-im/specs/pull/159#discussion_r1193581596 as not implemented.

> | Post in group | `/g/[group_key]/[message_key]` | Go to a message `message_key` in the group chat `group_key`

Reviewer: Is this example already supported? What does `group_key` consist of?

Reviewer: > I think that's removed and left to be revised/re-implemented post MVP. – https://github.com/status-im/specs/pull/159#issuecomment-1546914974

Reviewer: Removed at https://github.com/status-im/specs/pull/159#discussion_r1193567151

---
permalink: /spec/17
parent: Draft specs
title: 17/URL SCHEME
---

# 17/Defined Url Scheme

> Version: 0.1.0
>
> Status: Draft
>
> Authors: Aleksandar Djenic <aleksandardjenic@status.im>
>

## Table of Contents

 1. [Url Scheme](#url-scheme)
 2. [Defined urls](#defined-urls)

## Url scheme

- internal: “status-im://”

> - external: “https://join.status.im”

Reviewer 2: Why are you using [left](https://www.compart.com/en/unicode/U+201C) and [right](https://www.compart.com/en/unicode/U+201D) double quotes instead of [normal double quotes](https://www.compart.com/en/unicode/U+0022)?

Author: @jakubgs that's most likely case cause I have local languages added, and didn't pay attention when it switched. I fixed that part. Thanks.

## Defined urls

| Name | Short Url | Long Url | Description |
| ----- | ---- | ---- | ---- |
| Browse | `/b/[url]` | `/browse/[url]`  | Opens `url` in the app's browser |

> | User profile | `/u/[user_pk &#124; ens_name]` | `/user/[user_pk &#124; ens_name]` | Displays user profile popup for user with `user_pk` or `ens_name` |

Reviewer 2: What is this `&#124;`? Were you trying to add a vertical bar? Just write `or`.

Author: I couldn't use `|` in a cell and that's why I added it as html code. Changed it with `or`.

Reviewer 2: Yeah, that won't work in most markdown parsers.

| | `/u#<ens_name>` | |
| Community |	`/c/<data>#<signature>` | Open community |
| | `/c#<compressed_community_public_key>` | |
| Community channel | `/cc/[compressed_channel_key]`| Open community which has a channel with `compressed_channel_key` and makes that channel active |

> | Browse | `/b/[url]` |  Open `url` in the app's browser |
> | Post in channel | `/cc/[channel_key]/[message_key]` | Go to a message `message_key` in the channel `channel_key`

Reviewer: ```suggestion
```

---
permalink: /spec/17
parent: Draft specs
title: 17/URL SCHEME
---

# 17/Defined Url Scheme

> Version: 0.1.0
>
> Status: Draft
>
> Authors: Aleksandar Djenic <aleksandardjenic@status.im> Michal Iskierko <michal@status.im>
>

## Table of Contents

 1. [Url Scheme](#url-scheme)
 2. [Defined url](#defined-url)

## Url scheme

- internal: "status-app://"
- external: "https://status.app"

## Defined url

| Name | Url | Description |
| ----- | ---- | ---- |
| User profile | `/u/<data>#<signature>` | Display user |
| | `/u#<compressed_user_public_key>` | |
| | `/u#<ens_name>` | |
| Community |	`/c/<data>#<signature>` | Open community |
| | `/c#<compressed_community_public_key>` | |
| Community channel | `/cc/[compressed_channel_key]`| Open community which has a channel with `compressed_channel_key` and makes that channel active |
| Browse | `/b/[url]` |  Open `url` in the app's browser |
| Post in channel | `/cc/[channel_key]/[message_key]` | Go to a message `message_key` in the channel `channel_key`

> | Post in group | `/g/[group_key]/[message_key]` | Go to a message `message_key` in the group chat `group_key`
> 

Reviewer: ```suggestion
```
